### PR TITLE
ACS-2336 POST endpoints for archive/archive-restore content

### PR DESCRIPTION
**Pull request by mpichura**
This PR contains changes specific only to remote-api module assuming there is no downstream Java API implementation delivered yet.

### CODE REVIEW

**Comment by montgolfiere** [comment]
I suggest marking as Draft for now (until you/we can update with the actual Java API impl).

File: remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java [hunk 1]
```diff
@@ -74,6 +76,27 @@ public ContentStorageInfo getStorageInfo(String nodeId, String contentPropName,
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean requestRestoreContentFromArchive(String nodeId, String contentPropName,
+                                                    RestoreArchivedContentRequest restoreArchivedContentRequest)
+    {
+        //TODO: call downstream implementations (ContentService or equivalent) when added
```
Comment (PunchedChimera) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java:96]: TODO?

File: remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java [hunk 2]
```diff
@@ -72,4 +77,42 @@ public void afterPropertiesSet() throws Exception
...
+    @Operation("archive")
+    @WebApiDescription(title = "Request send content to archive",
+            description = "Submits a request to send content to archive",
+            successStatus = HttpServletResponse.SC_ACCEPTED)
+    public void requestArchiveContent(String entityResourceId, String id, Void voidRequest, Parameters parameters,
+                                      WithResponse withResponse)
+    {
+
+        final boolean result = storageInformation.requestArchiveContent(entityResourceId, id);
+        //TODO: handle below status codes and possible exceptions when downstream implementations are present
```
Comment (PunchedChimera) [inline remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java:91]: TODO? Ticket reference? Is this needed? Only exceptions I can for see are `ArchivedIOException` or `IllegalArgumentException` These should not need additional handling.

File: remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java [hunk 3]
```diff
@@ -72,4 +77,42 @@ public void afterPropertiesSet() throws Exception
...
+        if (result) {
+            withResponse.setStatus(HttpServletResponse.SC_ACCEPTED);
+        } else {
+            withResponse.setStatus(HttpServletResponse.SC_NOT_IMPLEMENTED);
+        }
```
Comment (PunchedChimera) [inline remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java:96]: formatting

File: remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java [hunk 4]
```diff
@@ -72,4 +77,42 @@ public void afterPropertiesSet() throws Exception
...
+                                                 RestoreArchivedContentRequest restoreArchivedContentRequest,
+                                                 Parameters parameters, WithResponse withResponse)
+    {
+        final boolean result = storageInformation.requestRestoreContentFromArchive(entityResourceId, id, restoreArchivedContentRequest);
+        //TODO: handle below status codes and possible exceptions when downstream implementations are present
+        if (result) {
+            withResponse.setStatus(HttpServletResponse.SC_ACCEPTED);
+        } else {
+            withResponse.setStatus(HttpServletResponse.SC_NOT_IMPLEMENTED);
+        }
```
Comment (PunchedChimera) [inline remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java:116]: formatting

**Review by PunchedChimera — COMMENTED** [review]
Looks good, just a couple of comment below:

File: remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java (hunk 1 above)
Comment (mpichura) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java]: Gone now (there's already a downstream implementation merged).

File: remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java (hunk 2 above)
Comment (mpichura) [reply remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java]: Thanks - removed that. Though I was thinking about `UnsupportedOperationException` versus a false return handling. 
Any other suggestion are welcomed.

File: remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java (hunk 3 above)
Comment (mpichura) [reply remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java]: Thanks - fixed.

File: remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java (hunk 4 above)
Comment (mpichura) [reply remote-api/src/main/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelation.java]: Thanks - fixed.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java [hunk 5]
```diff
@@ -74,6 +80,33 @@ public ContentStorageInfo getStorageInfo(String nodeId, String contentPropName,
...
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean requestRestoreContentFromArchive(String nodeId, String contentPropName,
+                                                    RestoreArchivedContentRequest restoreArchivedContentRequest)
+    {
+        final NodeRef nodeRef = new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, nodeId);
+        final QName propQName = getQName(contentPropName);
+        final Map<String, Serializable> restoreParams = restoreArchivedContentRequest.getRestorePriority() == null ?
```
Comment (PunchedChimera) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java:104]: Confused here. EXPIRY_DAYS seems to be missing. Both from this map and the restoreArchivedContentRequest. Wouldn't it be better to add a Map like the archiveContentRequest?

File: remote-api/src/main/java/org/alfresco/rest/api/model/RestoreArchivedContentRequest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+ * @author mpichura
+ */
+@Experimental
+public class RestoreArchivedContentRequest
+{
+    /**
+     * Restore priority. Possible values: Standard/High.
+     * This field is optional.
+     */
+    private String restorePriority;
```
Comment (PunchedChimera) [inline remote-api/src/main/java/org/alfresco/rest/api/model/RestoreArchivedContentRequest.java:44]: As above, confused here. Should we not have a Map<String, Serializable>? Or also include EXPIRY_DAYS?

**Review by PunchedChimera — CHANGES_REQUESTED** [review]
I just had a glance from my phone. Some clarity needed below. I'll take a more thorough look in the morning.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java (hunk 5 above)
Comment (PunchedChimera) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/ContentStorageInformationImpl.java:104]: Following conversation, I understand now that EXPIRY_DAYS are not being used as a requestParam as this does not allow the request to be cloud agnostic. Instead we are using config on the connector side.

File: remote-api/src/main/java/org/alfresco/rest/api/model/RestoreArchivedContentRequest.java (hunk 6 above)
Comment (PunchedChimera) [reply remote-api/src/main/java/org/alfresco/rest/api/model/RestoreArchivedContentRequest.java:44]: Following conversation, I understand now that EXPIRY_DAYS are not being used as a requestParam as this does not allow the request to be cloud agnostic. Instead we are using config on the connector side.

File: remote-api/src/test/java/org/alfresco/rest/api/impl/ContentStorageInformationImplTest.java [hunk 7]
```diff
@@ -26,26 +26,39 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
 @RunWith(MockitoJUnitRunner.class)
 public class ContentStorageInformationImplTest
 {
+    static final String DUMMY_NODE_ID = "dummy-node-id";
+    static final String CONTENT_PROP_NAME = "cm:content";
+    static final String NORMAL_PRIORITY = "Normal";
```
[the remark this answers is not in the document]
Comment (mpichura) [reply remote-api/src/test/java/org/alfresco/rest/api/impl/ContentStorageInformationImplTest.java:60]: Thanks, change coming.

File: remote-api/src/test/java/org/alfresco/rest/api/impl/ContentStorageInformationImplTest.java [hunk 8]
```diff
@@ -26,26 +26,39 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
 @RunWith(MockitoJUnitRunner.class)
 public class ContentStorageInformationImplTest
 {
+    static final String DUMMY_NODE_ID = "dummy-node-id";
```
[the remark this answers is not in the document]
Comment (mpichura) [reply remote-api/src/test/java/org/alfresco/rest/api/impl/ContentStorageInformationImplTest.java:58]: Thanks, change coming.

File: remote-api/src/test/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelationTest.java [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NodeStorageInfoRelationTest extends TestCase
+{
+    static final String DUMMY_NODE_ID = "dummy-node-id";
```
[the remark this answers is not in the document]
Comment (mpichura) [reply remote-api/src/test/java/org/alfresco/rest/api/nodes/NodeStorageInfoRelationTest.java:55]: Thanks, change coming.

**Review by PunchedChimera — APPROVED** [review]

**Review by montgolfiere — APPROVED** [review]

**Pull request merged**